src/client.js

```diff
@@ -305,427 +305,558 @@ class ConnectionParameters {
-                ]
-            }
-        };
-        // Convert to current representation
-        queryDef = client._convertToRepresentation(queryDef, "SimpleJson");
-        const query = client.NLWS.xtkQueryDef.create(queryDef);
-        const extAccount = await query.executeQuery();
-
-        const cipher = await client._getSecretKeyCipher();
-        const type = EntityAccessor.getAttributeAsLong(extAccount, "type");
-        if (type == 3) {
-            // Mid-souring
-            const endpoint = EntityAccessor.getAttributeAsString(extAccount, "server");
-            const user = EntityAccessor.getAttributeAsString(extAccount, "account");
-            const password = cipher.decryptPassword(EntityAccessor.getAttributeAsString(extAccount, "password"));
-            return ConnectionParameters.ofUserAndPassword(endpoint, user, password, client._connectionParameters._options);
+      var queryDef = {
+        "schema": "nms:extAccount",
+        "operation": "get",
+        "select": {
+          "node": [
+            { "expr": "@id" },
+            { "expr": "@name" },
+            { "expr": "@label" },
+            { "expr": "@type" },
+            { "expr": "@account" },
+            { "expr": "@password" },
+            { "expr": "@server" }
+          ]
+        },
+        "where": {
+          "condition": [
+            { "expr": client.sdk.escapeXtk`@name=${extAccountName}` },
+            { "expr": "@type=3" }
+          ]
         }
-        throw CampaignException.CREDENTIALS_FOR_INVALID_EXT_ACCOUNT(extAccountName, type);
+      };
+        // Convert to current representation
+      queryDef = client._convertToRepresentation(queryDef, "SimpleJson");
+      const query = client.NLWS.xtkQueryDef.create(queryDef);
+      const extAccount = await query.executeQuery();
+
+      const cipher = await client._getSecretKeyCipher();
+      const type = EntityAccessor.getAttributeAsLong(extAccount, "type");
+      if (type == 3) {
+        // Mid-souring
+        const endpoint = EntityAccessor.getAttributeAsString(extAccount, "server");
+        const user = EntityAccessor.getAttributeAsString(extAccount, "account");
+        const password = cipher.decryptPassword(EntityAccessor.getAttributeAsString(extAccount, "password"));
+        return ConnectionParameters.ofUserAndPassword(endpoint, user, password, client._connectionParameters._options);
+      }
+      throw CampaignException.CREDENTIALS_FOR_INVALID_EXT_ACCOUNT(extAccountName, type);
     }
 
-}
+  }
 
-// ========================================================================================
-// ACC Client
-// ========================================================================================
+  // ========================================================================================
+  // File Uploader
+  // ========================================================================================
+
+  const fileUploader = (client) => {
```

Comment:
Is there a particular reason not to use a class as this is done everywhere else in the SDK?

Reply:
SDK follows both of the approaches. 
For example:
`const clientHandler = (representation, headers, pushDownOptions) => {` 
and 
`class ConnectionParameters {`

i just picked one which suits better for this use case.

Reply:
Good point. clientHandler should also use classes, I'll change it.
Can you change to use class as well?
Maybe I'm missing something, but I'm not sure I understand why using a function is better here. It seems to me it's harder to read